### PR TITLE
Offline Mode: Media upload status view

### PR DESCRIPTION
**Pull request by kean**
This PR adds a media upload status view to surface the uploads information to both users and developers. It also changes the (new) UI/UX for managing media uploads prior to publishing. In the previous implementation, on any upload failure, the app would instantly show a "Retry" button that you had to tap manually - awkward UX. Now, the retries happen automatically, and instantly in case of the re-connects.

## To test:

- Enable Airplane mode
- Create a draft with media
- Tap "Publish"
- Open Upload Status view
- ✅  Verify that the upload status is "Pending"
- Disable Airplane mode
- ✅  Verify that the upload restated and completed automatically 

## Known Issues

There is something going on with the network protocols / uploads where status is now always reported accurately.

## Regression Notes
1. Potential unintended areas of impact: Publishing 
2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Review by momo-ozawa — COMMENTED** [review]
@kean 👋 

I think the tap area for the info icon on the publish button is too small. When I test on my actual iPad/iPhone, it's very hard to get it to show the upload status screen.

Also, what do you think of removing the progress bar below the navigation title? I don't think it provides any extra information because you can see the individual upload progress for each media item.

**Comment by kean** [comment]
> what do you think of removing the progress bar below the navigation title?

The idea was to show the overall progress, but it does feel a bit excessive.

**Comment by kean** [comment]
I'm considering adding it to the editor as well. This way, if you see your draft is syncing, you could open an editor to check the status.

**Comment by kean** [comment]
I removed the progress bar and made the entire "Publish" button tappable (now also has a highlight on tap).

File: WordPress/Classes/ViewRelated/Post/PostMediaUploadViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+private extension Media {
+    var isUploadNeeded: Bool {
+        mediaID == nil || mediaID?.intValue == 0
+    }
+}
+
+private enum Strings {
+    static func progress(completed: Int64, total: Int64) -> String {
+        let format = NSLocalizedString("postMediaUploadStatusView.progress", value: "%@ of %@", comment: "Shows the upload progress with two parameters: preformatted completed and total bytes")
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Post/PostMediaUploadViewModel.swift]: Let's use positional placeholders (ref: pbMoDN-1Df-p2) 
```suggestion
        let format = NSLocalizedString("postMediaUploadStatusView.progress", value: "%1$@ of %2$@", comment: "Shows the upload progress with two preformatted parameters: %1$@ is the placeholder for completed bytes, and %2$@ is the placeholder for total bytes")
```

File: WordPress/Classes/ViewRelated/Post/Prepublishing/PublishButton.swift [hunk 2]
```diff
@@ -72,6 +82,11 @@ struct PublishButton: View {
         }
     }
 
+    @ViewBuilder
+    private func makeUploadingView(title: String, progress: PublishButtonState.Progress) -> some View {
+
+    }
+
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Post/Prepublishing/PublishButton.swift]:
```suggestion
```

File: WordPress/Classes/ViewRelated/Post/PostMediaUploadStatusView.swift [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                    .foregroundStyle(.secondary)
+                    .lineLimit(1)
+            }
+
+            Spacer()
+
+            switch viewModel.state {
+            case .uploaded:
+                Image(systemName: "checkmark.circle.fill")
+                    .foregroundStyle(.secondary.opacity(0.5))
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Post/PostMediaUploadStatusView.swift:52]: Wdyt of applying a green tint to the check mark icon? I think it makes it more obvious it's done.

 | 
-- | --


```suggestion
                    .foregroundColor(Color(uiColor: .success))
```

**Review by momo-ozawa — APPROVED** [review]
A few minor comments, but looks good otherwise!

File: WordPress/Classes/ViewRelated/Post/PostMediaUploadStatusView.swift (hunk 3 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Post/PostMediaUploadStatusView.swift:52]: The progress indicators are green, so I wanted something less eye-catching. It's similar to how Apple Music shows download indicators.

Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Post/PostMediaUploadStatusView.swift:52]: Hmmm, okay. I see your point - the podcasts app does it too.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**